### PR TITLE
GitQuery 3.0.4

### PR DESCRIPTION
**Pull request by tinder-aminghadersohi**
- Fixed the bug that was writing absolute paths for repoDir and outputDir
`repoDir: /Users/aminghadersohi/Tinder/TinderBackend/events.sdk/build/tmp/.gitquery`
should be
`repoDir: build/tmp/.gitquery`
- Simplify constructor call for Yaml to avoid this error: `org.yaml.snakeyaml.representer.Representer.<init>(Lorg/yaml/snakeyaml/DumperOptions;)V`
- Upgrade to Snakeyaml 1.27 
- new Groovy dsl example
- Allow init-config to use includeGlobs in config file. This means we don't have to duplicate them in the gitQueryInit gradle extension

### CODE REVIEW

File: samples/protobuf-sync-code-gen/gitquery.yml [hunk 1]
```diff
@@ -13,8 +13,8 @@ initConfig:
   includeGlobs:
   - '**/examples/addressbook.proto'
   revision: v3.14.0
-outputDir: /Users/aming/github/Tinder/GitQuery/samples/protobuf-sync-code-gen/src/main/proto
+outputDir: src/main/proto
```
Comment (tinder-aminghadersohi) [inline samples/protobuf-sync-code-gen/gitquery.yml:16]: see the fix here to the example

File: samples/protobuf-sync-groovy-dsl/build.gradle [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    repoDir = "tmp/.gitquery"
+}
+
+// This will init/update the config file every time the gitQueryInit task runs.
+gitQueryInit {
+    flatFiles = false
+    includeGlobs = ["**/examples/addressbook.proto"]
+    excludeGlobs = ["**/generated/**/*.proto"]
+    revision = "v3.14.0"
+}
```
Comment (tinder-aminghadersohi) [inline samples/protobuf-sync-groovy-dsl/build.gradle:22]: groovy syntax is the same as android

File: core/src/main/kotlin/com/tinder/gitquery/core/config/GitQueryConfig.kt [hunk 3]
```diff
@@ -78,16 +78,16 @@ data class GitQueryConfig(
         val writer = FileWriter(filename)
         val options = DumperOptions()
         options.defaultFlowStyle = DumperOptions.FlowStyle.BLOCK
-        val yaml = Yaml(Constructor(GitQueryConfig::class.java), Representer(options), options)
+        val yaml = Yaml(options)
```
Comment (tinder-aminghadersohi) [inline core/src/main/kotlin/com/tinder/gitquery/core/config/GitQueryConfig.kt:81]: bug fix

File: samples/protobuf-sync-code-gen/gitquery.yml (hunk 1 above)
Comment (Tinder-KevinZhu) [reply samples/protobuf-sync-code-gen/gitquery.yml:16]: Ah yeah i was wondering whether the /Users/aming/github portion of the output directory was intended

File: samples/protobuf-sync-code-gen/build.gradle.kts [hunk 4]
```diff
@@ -9,26 +9,17 @@ import com.google.protobuf.gradle.protoc
 plugins {
     application
     id("com.google.protobuf") version "0.8.13"
-    id("com.tinder.gitquery") version "3.0.0"
+    id("com.tinder.gitquery") version "3.0.4"
```
Comment (Tinder-KevinZhu) [inline samples/protobuf-sync-code-gen/build.gradle.kts:12]: 3.0.4? I thought it was 3.0.3 with these new commits

File: samples/protobuf-sync-code-gen/build.gradle.kts [hunk 5]
```diff
@@ -9,26 +9,17 @@ import com.google.protobuf.gradle.protoc
...
 gitQuery {
     autoSync = true
-    cleanOutput = true
     configFile = "gitquery.yml"
     outputDir = protoDir
-    repoDir = "${project.buildDir}/tmp/.gitquery"
-}
-
-// This will init/update the config file every time the gitQuery task runs.
-gitQueryInit {
```
Comment (Tinder-KevinZhu) [inline samples/protobuf-sync-code-gen/build.gradle.kts:27]: why the deletion of the gitQueryInit example?

**Review by Tinder-KevinZhu — APPROVED** [review]

**Pull request merged**